esp/esp/users/migrations/0020_contactinfo_cleanup.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.29 on 2020-10-08 20:04
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+def set_my_defaults(apps, schema_editor):
+    ContactInfo = apps.get_model('users', 'ContactInfo')
+    for ci in ContactInfo.objects.all():
+        if ci.user is None:
+            # Look for associated registration profiles with user info
+            if ci.as_user.exists():
+                ci.user = ci.as_user.latest('last_ts').user
+                ci.save()
+            elif ci.as_guardian.exists():
+                ci.user = ci.as_guardian.latest('last_ts').user
+                ci.save()
+            elif ci.as_emergency.exists():
+                if not (ci.first_name and ci.last_name):
+                    # We want to delete emergency contact info for non-students
```

Comment:
Don't see the relationship between the code and the comment. How is having (or not having) a first name and last name the same as being (or not being) a student?

Reply:
Ah, so I'm trying to retroactively fix the problem caused by this old line (now moved into the student forloop):
https://github.com/learning-unlimited/ESP-Website/blob/0fa4528f5c484de79569d651f4dea0672214444a/esp/esp/web/views/myesp.py#L174.
We used to add a `contact_emergency` for all users, not just students. But we only ask for emergency contact info in the student version of the profile form, so it always resulted in a blank contactinfo. So here we are looking to see if a contact info has been used for `contact_emergency` but does not have a first and last name (indicating it was created by the above line for a non-student).